File: client-android/lib/src/main/kotlin/stasis/client_android/lib/ops/recovery/RecoverySourceKind.kt
package stasis.client_android.lib.ops.recovery

import stasis.client_android.lib.collection.rules.SourceUri

sealed interface RecoverySourceKind {
    data object Filesystem : RecoverySourceKind

    data class Library(val scheme: String) : RecoverySourceKind

    companion object {
        fun forEntity(entity: String): RecoverySourceKind =
            SourceUri.scheme(entity)?.let { Library(it) } ?: Filesystem
    }
}

File: client-ios/lib/Sources/StasisClientLib/Ops/Recovery/RecoverySourceKind.swift
public enum RecoverySourceKind: Sendable, Hashable {
    case filesystem
    case library(scheme: String)

    public static func forEntity(_ entity: String) -> RecoverySourceKind {
        if let scheme = SourceUri.scheme(entity) {
            return .library(scheme: scheme)
        }
        return .filesystem
    }
}
